src/main/java/com/amazon/opendistroforelasticsearch/security/configuration/DlsFlsValveImpl.java

```diff
@@ -253,4 +260,58 @@ public void accept(StringTerms.Bucket bucket) {
             this.bucket = bucket;
         }
     }
+
+    private static class StringTermsGetter {
+        private static final Field REDUCE_ORDER = getField(InternalTerms.class, "reduceOrder");
+        private static final Field TERM_BYTES = getField(StringTerms.Bucket.class, "termBytes");
+        private static final Field FORMAT = getField(InternalTerms.Bucket.class, "format");
+
+        private StringTermsGetter() {
+        }
+
+        private static <T> Field getFieldPrivileged(Class<T> cls, String name) {
+            try {
+                final Field field = cls.getDeclaredField(name);
+                field.setAccessible(true);
+                return field;
+            } catch (NoSuchFieldException | SecurityException e) {
+                log.error("Failed to get class {} declared field {}", cls.getSimpleName(), name, e);
+                if (e instanceof RuntimeException) {
+                    throw (RuntimeException) e;
+                } else {
+                    throw new RuntimeException(e);
+                }
+            }
+        }
+
+        private static <T> Field getField(Class<T> cls, String name) {
+            SpecialPermission.check();
+            return AccessController.doPrivileged((PrivilegedAction<Field>) () -> getFieldPrivileged(cls, name));
+        }
+
+        private static <T, C> T getFieldValue(Field field, C c) {
+            try {
+                return (T)field.get(c);
+            } catch (IllegalArgumentException | IllegalAccessException e) {
+                log.error("Exception while getting value {} of class {}", field.getName(), c.getClass().getSimpleName(), e);
+                if (e instanceof RuntimeException) {
+                    throw (RuntimeException) e;
```

Comment:
Cant it be handled the same way as "throw new RuntimeException(e)" ?

Reply:
There is no need to wrap `RuntimeException` in another `RuntimeException`